packages/eslint-plugin/src/rules/boolean-literal-compare.ts

```diff
@@ -0,0 +1,136 @@
+import { TSESTree } from '@typescript-eslint/typescript-estree';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+import { getEqualsKind } from '../util';
+
+export default util.createRule({
+  name: 'boolean-literal-compare',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Flags unnecessary equality comparisons against booleans.',
+      tslintName: 'no-boolean-literal-compare',
+      category: 'Best Practices',
+      recommended: 'error',
```

Comment:
Cannot recommend without a breaking change
```suggestion
      recommended: false,
```